classes/Helper.php

```diff
@@ -0,0 +1,122 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Version information
+ *
+ * @package    report_rolessitemap
+ * @copyright  2022 Andreas Schenkel
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace report_rolessitemap;
+use moodle_url;
+use report_rolessitemap\Misc;
+
+defined('MOODLE_INTERNAL') || die();
+
+class Helper {
+    /**
+     * Shows all categories and the assigned users with their roles
+     *
+     * @return void
+     * @throws \dml_exception
+     */
+    public function rendercategoricesandroles(): void {
+        GLOBAL $DB, $OUTPUT;
+        $systemcontext = \context_system::instance();
+        $roles = role_fix_names(get_all_roles(), $systemcontext, ROLENAME_ORIGINAL);
+        //$roles = $DB->get_records('role', null, 'id', 'id, shortname');
+        $categorieslist = \core_course_category::make_categories_list();
+        // Now populate $categorieslistandroles with the information to be rendered using mustach-template.
+        $categorieslistandroles = [];
+        foreach ($categorieslist as $categoryid => $categoryname) {
+            $context = \context_coursecat::instance($categoryid);
+            $params = array('contextid' => $context->id);
```

Comment:
New array syntax could be used here:

```suggestion
            $params = [];
            $params['contextid'] = $context->id;
```

Reply:
fixed